gpustack/routes/proxy.py

```diff
@@ -81,6 +77,32 @@ async def proxy(request: Request, url: str):
             )
 
 
+def validate_http_method(method: str):
+    allowed_methods = ["GET", "POST", "PUT", "DELETE"]
+    if method not in allowed_methods:
+        raise BadRequestException(message=f"HTTP method '{method}' is not allowed")
+
+
+def validate_url(url: str):
+    if not url:
+        raise BadRequestException(message="Missing 'url' query parameter")
+
+    try:
+        parsed_url = urlparse(url)
+    except Exception:
```

Comment:
urlparse does not raise exceptions, check parsed_url instead

Reply:
urlparse raise exception in some case, for example: `urlparse('//[oops')`